### ISSUE TITLE
Use Google Test or similar testing framework

### ISSUE THREAD

**Issue by liuzicheng1987** [getml/reflect-cpp#34]
Pertaining to the discussion related to @zerolfx's great PR (https://github.com/getml/reflect-cpp/pull/29), it is necessary to use some kind of testing framework to fully reap the benefits of Github Actions. We are looking to introduce more serialization formats over the next 2-3 weeks. Being able to test them all in an automated manner will be a great benefit that I really look forward to.

**Comment by ChemiCalChems** [comment]
For starters, we don't really need to move to Google Test right away. Making sure our current tests return `true` or `false` so the test process can return `0` on success and non-zero on failure is enough to get the GitHub Actions saga rolling. 

I'm not saying I'm against introducing Google Test (in fact I was quite surprised when I first looked at the project and I saw no test library at all being used), I'm just saying as it stands, a quick PR changing a few things in our tests is enough for GitHub Actions for the time being.
